File: the99thTime/Programs_v2/MonteCarloSimulationBenchmark.py
import random
import datetime

NUM_POINTS = 36000000

def simulateMonteCarlo(num_points):
    pointsInsideCircle = 0

    for _ in range(num_points):
        x = random.random()
        y = random.random()

        if is_inside_circle(x, y):
            pointsInsideCircle += 1

    return pointsInsideCircle

def is_inside_circle(x, y):
    distance = (x ** 2 + y ** 2) ** 0.5
    return distance <= 1.0

def calculate_pi_approximation(pointsInsideCircle, num_points):
    return 4.0 * pointsInsideCircle / num_points

def main():
    current_datetime = datetime.datetime.now()
    date_time_string = current_datetime.strftime("%Y%m%d_%H%M%S")

    pi_approximation = calculate_pi_approximation(simulateMonteCarlo(NUM_POINTS), NUM_POINTS)

    with open(f"Output/montecarloResult_{date_time_string}.txt", "w") as file:
        file.write(str(pi_approximation))

if __name__ == "__main__":
    main()

File: the99thTime/Programs_v2/MonteCarloSimulationBenchmark.java
import java.util.Random;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class MonteCarloSimulationBenchmark {
    private static final int NUM_POINTS = 36000000;

    public static void main(String[] args) {

        LocalDateTime currentDateTime = LocalDateTime.now();

        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss");
        String dateTimeString = currentDateTime.format(formatter);

        int pointsInsideCircle = simulateMonteCarlo(NUM_POINTS);

        double piApproximation = calculatePiApproximation(pointsInsideCircle, NUM_POINTS);
  
        try (PrintWriter writer = new PrintWriter(new FileWriter("Output/montecarloResult_" + dateTimeString + ".txt"))) {
            writer.println(piApproximation);
        } catch (IOException e) {
            e.printStackTrace();
        }
        System.exit(1) ;
    }

    private static int simulateMonteCarlo(int numPoints) {
        Random random = new Random();
        int pointsInsideCircle = 0;

        for (int i = 0; i < numPoints; i++) {
            double x = random.nextDouble();
            double y = random.nextDouble();

            if (isInsideCircle(x, y)) {
                pointsInsideCircle++;
            }
        }

        return pointsInsideCircle;
    }

    private static boolean isInsideCircle(double x, double y) {
        double distance = Math.sqrt(x * x + y * y);
        return distance <= 1.0;
    }

    private static double calculatePiApproximation(int pointsInsideCircle, int numPoints) {
        return 4.0 * pointsInsideCircle / numPoints;
    }
}
